db/keyval/redis/connection.go

```diff
@@ -130,15 +69,310 @@ func createTLSConfig(config TLS) (*tls.Config, error) {
 	return tlsConfig, nil
 }
 
-// GenerateConfig Generates a yaml file using the given configuration object
-func GenerateConfig(config interface{}, path string) error {
-	bytes, err := yaml.Marshal(&config)
+///////////////////////////////////////////////////////////////////////////////
+// go-redis https://github.com/go-redis/redis
+
+// GoRedisNil go-redis return this error when Redis replies nil, .e.g. when key does not exist.
+const GoRedisNil = goredis.Nil
+
+// Client Common interface to adapt all types of Redis clients
+type Client interface {
+	// The easiest way to adapt Cmdable interface is to just embed it.
+	goredis.Cmdable
+	/*
+		But that means we'll have to mock each and every method in Cmdable for unit tests,
+		making it a whole lot more complicated.  When the time comes, it may be more manageable
+		to only delcare (duplicate) the methods we need from Cmdable.  As follows:
+			Del(keys ...string) *goredis.IntCmd
+			Get(key string) *goredis.StringCmd
+			MGet(keys ...string) *goredis.SliceCmd
+			MSet(pairs ...interface{}) *goredis.StatusCmd
+			Scan(cursor uint64, match string, count int64) *goredis.ScanCmd
+			Set(key string, value interface{}, expiration time.Duration) *goredis.StatusCmd
+	*/
+
+	// Declare these additional methods to enable access to them through this interface
+	Close() error
+	TxPipeline() goredis.Pipeliner
+	PSubscribe(channels ...string) *goredis.PubSub
+}
+
+// ClientConfig Configuration common to all types of Redis clients
+type ClientConfig struct {
```

Comment:
Could you please move config to separate file config.go (like Lukas did in etcdv3). I have asked for that also Madhura (we could have a convention like this).